x/oracle/keeper/grpc_query_test.go

```diff
@@ -137,8 +140,123 @@ func (s *IntegrationTestSuite) TestQuerier_AggregateVotes() {
 	s.Require().Equal([]types.AggregateExchangeRateVote(nil), res.AggregateVotes)
 }
 
+func (s *IntegrationTestSuite) TestQuerier_AggregateVoteInvalidExchangeRate() {
+	res, err := s.queryClient.AggregateVote(s.ctx.Context(), &types.QueryAggregateVoteRequest{
+		ValidatorAddr: valAddr.String(),
+	})
+	s.Require().Nil(res)
+	s.Require().ErrorContains(err, "no aggregate vote")
+}
+
+func (s *IntegrationTestSuite) TestQuerier_AggregatePrevoteInvalidExchangeRate() {
+	res, err := s.queryClient.AggregatePrevote(s.ctx.Context(), &types.QueryAggregatePrevoteRequest{
+		ValidatorAddr: valAddr.String(),
+	})
+	s.Require().Nil(res)
+	s.Require().ErrorContains(err, "no aggregate prevote")
+}
+
 func (s *IntegrationTestSuite) TestQuerier_Params() {
 	res, err := s.queryClient.Params(context.Background(), &types.QueryParamsRequest{})
 	s.Require().NoError(err)
 	s.Require().Equal(types.DefaultGenesisState().Params, res.Params)
 }
+
+func (s *IntegrationTestSuite) TestQuerier_ExchangeRatesInvalidExchangeRate() {
+	resExchangeRate, err := s.queryClient.ExchangeRates(s.ctx.Context(), &types.QueryExchangeRatesRequest{
+		Denom: " ",
+	})
+	s.Require().Nil(resExchangeRate)
+	s.Require().ErrorContains(err, "unknown denom")
+}
+
+func (s *IntegrationTestSuite) TestQuerier_AggregatePrevoteInvalidValAddr() {
+	resExchangeRate, err := s.queryClient.AggregatePrevote(s.ctx.Context(), &types.QueryAggregatePrevoteRequest{
+		ValidatorAddr: "valaddrInvalid",
+	})
+	s.Require().Nil(resExchangeRate)
+	s.Require().ErrorContains(err, "decoding bech32 failed")
+}
+
+func (s *IntegrationTestSuite) TestQuerier_AggregatePrevotesAppendVotes() {
+	s.app.OracleKeeper.SetAggregateExchangeRatePrevote(s.ctx, valAddr, types.NewAggregateExchangeRatePrevote(
+		types.AggregateVoteHash{},
+		valAddr,
+		uint64(s.ctx.BlockHeight()),
+	))
+
+	_, err := s.queryClient.AggregatePrevotes(s.ctx.Context(), &types.QueryAggregatePrevotesRequest{})
+	s.Require().Nil(err)
+}
+
+func (s *IntegrationTestSuite) TestQuerier_AggregateVotesAppendVotes() {
+	s.app.OracleKeeper.SetAggregateExchangeRateVote(s.ctx, valAddr, types.NewAggregateExchangeRateVote(
+		types.DefaultGenesisState().ExchangeRates,
+		valAddr,
+	))
+
+	_, err := s.queryClient.AggregateVotes(s.ctx.Context(), &types.QueryAggregateVotesRequest{})
+	s.Require().Nil(err)
+}
+
+func TestEmptyRequest(t *testing.T) {
```

Comment:
We should use the test suite here : 
```suggestion
func (s *IntegrationTestSuite) TestEmptyRequest() {
```

Reply:
I believe here it's needed to send in `s.ctx.Context()`, and it's just uniform